server/smart_auth.go

```diff
@@ -0,0 +1,57 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/labstack/echo"
+)
+
+// SmartAuthHandler middleware provides authorization checking aginst the scopes
+// described in the SMART App Authorization Guide
+// http://docs.smarthealthit.org/authorization/
+//
+// The assumption is that there is an API Gatway that is deployed in front of
+// this go application which handles OAuth 2 token introspection and then
+// provides information about the request in HTTP headers.
+//
+// This code assumes that two headers will be present in the HTTP request
+// X-DELEGATED - This will be set to "true" when the request is being made by
+// a client on behalf of a user. If this header is not present or any value
+// other than "true", this middleware will pass the request through.
```

Comment:
Oh!  This surprises me.  So if SmartAuth is on, but we don't pass an X-DELEGATED header, then we just let it on through?  I totally expected it to reject any request that didn't contain authorization info...  Is this part of the spec?

It just seems to me that if you enable authorization, then you want it to be as strict as possible -- no?  Or is this so you can use a firewall to block the "unprotected" port but still be able to access it without auth from the local machine?